lisa/analysis/idle.py

```diff
@@ -236,6 +236,27 @@ def df_cluster_idle_state_residency(self, cluster):
         idle_time_df.index.name = 'idle_state'
         return idle_time_df
 
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    @requires_events('cpu_idle')
+    def idle_cpus_at(self, timestamp):
+        """
+        :returns: list of CPUs that were idle at the given ``timestamp``
+        """
+        idles = []
+        for cpu in cpus:
+            df_active = trace.analysis.idle.signal_cpu_active(cpu)
+            before, after = df_float_index_neighbours(df_active, timestamp)
+
+            # If the CPU went idle before T, it was idle at T
+            # If the CPU left idle after T, it was idle at T
+            # Explicitely check value in case it's None
+            if before == False or after == True:
```

Comment:
Well the good thing is we get away with a single check. The way I see it a separate `None` check would be
```
if after is not None:
    idle |= after
if before is not None:
    idle |= not before

if idle:
    idles.append(cpu)
```

Because both `after` or `before` can be `None`, it's somewhat annoying to check that in a single if. I think the explicit check is fine (and it's commented!), but if you're adamant about it I'll split it out.

Reply:
The issue is that the only type known for sure here is `None`. However, it seems that `True == 1.0` and `False == 0.0` as well, so Python seems to diverges from its usual strong typing behavior and allows comparing any numeric type (bool is a subclass of int). Given that, your code should be robust against type change.

Reply:
Completely forgot about the types in those columns, you're right.